spec/requests/api/groups_spec.rb

```diff
@@ -0,0 +1,264 @@
+# - Creating multiple groups              /api/groups                           action "create"
+# - Edit a group                          /api/groups/:id                       action "edit"
+# - Edit multiple groups                  /api/groups                           action "edit"
+# - Delete a group                        /api/groups/:id                       DELETE
+# - Delete a group by action              /api/groups/:id                       action "delete"
+# - Delete multiple groups                /api/groups                           action "delete"
+#
+require 'spec_helper'
+
+describe ApiController do
+  include Rack::Test::Methods
+
+  let(:expected_attributes) { %w(id guid description group_type tenant_id) }
+
+  let(:sample_group1) { {:description => "sample_group_1"} }
+  let(:sample_group2) { {:description => "sample_group_2"} }
+  let(:group1) { FactoryGirl.create(:miq_group, sample_group1) }
+  let(:group2) { FactoryGirl.create(:miq_group, sample_group2) }
+
+  let(:role3)    { FactoryGirl.create(:miq_user_role) }
+  let(:tenant3)  { FactoryGirl.create(:tenant, :name => "Tenant2") }
+
+  before(:each) do
+    init_api_spec_env
+  end
+
+  def app
+    Vmdb::Application
+  end
+
+  describe "groups create" do
+    it "rejects creation without appropriate role" do
+      api_basic_authorize
+
+      run_post(groups_url, sample_group1)
+
+      expect_request_forbidden
+    end
+
+    it "rejects group creation with id specified" do
+      api_basic_authorize collection_action_identifier(:groups, :create)
+
+      run_post(groups_url, "description" => "sample group", "id" => 100)
+
+      expect_bad_request(/Invalid attribute/i)
+    end
+
+    it "rejects group creation with invalid role specified" do
+      api_basic_authorize collection_action_identifier(:groups, :create)
+
+      run_post(groups_url, "description" => "sample group", "role" => {"id" => 999_999})
+
+      expect_resource_not_found
+    end
+
+    it "rejects group creation with invalid tenant specified" do
+      api_basic_authorize collection_action_identifier(:groups, :create)
+
+      run_post(groups_url, "description" => "sample group", "tenant" => {"id" => 999_999})
+
+      expect_resource_not_found
+    end
+
+    it "rejects group creation with invalid filters specified" do
+      api_basic_authorize collection_action_identifier(:groups, :create)
+
+      run_post(groups_url, "description" => "sample group", "filters" => {"bogus" => %w(f1 f2)})
+
+      expect_bad_request(/Invalid filter/i)
+    end
+
+    it "supports single group creation" do
+      api_basic_authorize collection_action_identifier(:groups, :create)
+
+      run_post(groups_url, sample_group1)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_attributes)
+
+      group_id = @result["results"].first["id"]
+      expect(MiqGroup.exists?(group_id)).to be_true
```

Comment:
To be RSpec 3+ compatible, I think these would need to be `be true` (no underscore)